client/js/lounge.js

```diff
@@ -332,12 +332,26 @@ $(function() {
 		var text = msg.find(".text");
 
 		if (template === "msg_action") {
+			if (data.msg.invited !== undefined) {
+				data.msg.target = data.msg.invited;
+			} else if (data.msg.new_nick !== undefined) {
+				data.msg.target = data.msg.new_nick;
+			} else if (data.msg.whois !== undefined) {
+				data.msg.target = data.msg.whois.nick;
+			}
+
 			text.html(templates.actions[type](data.msg));
 		}
 
 		if ((type === "message" || type === "action") && chan.hasClass("channel")) {
 			var nicks = chan.find(".users").data("nicks");
 			if (nicks) {
+				nicks.forEach(function(nick) {
```

Comment:
I don't see any other way to do this, since you need to search in text for each user.